testsuite/classlibrary/TestUGen_initSample.sc

```diff
@@ -0,0 +1,253 @@
+		};
+		synthFn.loadToFloatArray(runtimeSeconds, server) { |out|
+			synthOutput = out;
+			cond.signalOne();
+		};
+		cond.waitFor(timeout) { synthOutput.notNil };
+		^synthOutput;
+	}
+
+	assertSynthOutput { |synthFn, targetOutput, timeout=1|
+		var nSamples = targetOutput.size;
+		// var synthFnCode = synthFn.def.sourceCode.replaceRegexp("[ \n\r\t]","");
+		var synthFnCode = synthFn.def.sourceCode;
+		var out = this.awaitSynthOutput(synthFn, nSamples);
+		if (out.isNil) {
+			this.assert(false, "% timed out (timeout: %)".format(synthFnCode, timeout))
+		} {
+
+			this.assertArrayFloatEquals(out.keep(nSamples), targetOutput,
+				"% should produce expected output".format(synthFnCode)
+			);
+		};
+	}
+
+	assertInitEqualsFirstSample { |synthFn|
+		var out = this.awaitSynthOutput({
+			var in = synthFn.value;
+			var initSample = if (in.rate == \audio) {
+				InitializationSample.ar(in)
+			} {
+				InitializationSample.ir(in)
+			};
+			[initSample, in]
+		}, 1);
+		var fnCode = synthFn.def.sourceCode.replaceRegexp("[ \n\r\t]","");
+		this.assertFloatEquals(out[0], out[1],
+			"% init sample and first sample should be equal".format(fnCode));
+	}
+
+	// Tests
+
+	test_InitializationSample {
+		this.assertSynthOutput(
+			{ InitializationSample.ar(1) }, [1.0, 1.0],
+			"InitializationSample"
+		);
+		this.assertSynthOutput(
+			{ InitializationSample.ar(0) }, [0.0, 0.0],
+			"InitializationSample"
+		);
+		this.assertInitEqualsFirstSample { InitializationSample.ar(1) };
+		this.assertInitEqualsFirstSample { InitializationSample.ar(0) }
+	}
+
+	test_TriggerUGens {
+		var basicTest = { |ugenClass|
+			var tests = [];
+			if(ugenClass.respondsTo(\ar)) {
+				tests = tests ++ [
+					"{%.ar(DC.ar(0))}",
+					"{%.ar(DC.ar(1))}",
+				]
+			};
+			if(ugenClass.respondsTo(\kr)) {
+				tests = tests ++ [
+					"{%.kr(DC.kr(0))}",
+					"{%.kr(DC.kr(1))}",
+				]
+			};
+			tests.do {|fn|
+				fn = fn.format(ugenClass).interpret;
+				this.assertInitEqualsFirstSample(fn)
+			}
+		};
+
+		basicTest.(Trig1);
+		basicTest.(Trig);
+		basicTest.(ToggleFF);
+		basicTest.(SetResetFF);
+		basicTest.(Latch);
+		basicTest.(Gate);
+		basicTest.(Schmidt);
+
+		basicTest.(PulseDivider);
+		// test first trigger is not ignored
+		this.assertSynthOutput(
+			{ PulseDivider.ar(Impulse.ar(SampleRate.ir/2), 1) },
+			[1,0,1,0,1,0,1,0],
+			"PulseDivider.ar(Impulse.ar(SampleRate.ir/2), 1)"
+		);
+		// superfluous? PulseDivider has only one calcFunc
+		this.assertSynthOutput(
+			{ PulseDivider.kr(Impulse.kr(ControlRate.ir/2), 1) },
+			[1,0,1,0,1,0,1,0],
+			"PulseDivider.kr(Impulse.kr(ControlRate.ir/2), 1)"
+		);
+
+		basicTest.(PulseCount);
+		basicTest.(Stepper);
+		this.assertInitEqualsFirstSample({Stepper.ar(DC.ar(1), DC.ar(1))});
+		this.assertInitEqualsFirstSample({Stepper.ar(DC.ar(1), DC.ar(0))});
+		this.assertInitEqualsFirstSample({Stepper.ar(DC.ar(0), DC.ar(1))});
+		this.assertInitEqualsFirstSample({Stepper.ar(DC.ar(0), DC.ar(0))});
```

Comment:
minor formatting: spaces within { }

There are more occurrences below, maybe just do a pass over the full file for formatting